components/esp_lvgl_port/CMakeLists.txt

```diff
@@ -1,17 +1,34 @@
 include($ENV{IDF_PATH}/tools/cmake/version.cmake) # $ENV{IDF_VERSION} was added after v4.3...
 
 if("${IDF_VERSION_MAJOR}.${IDF_VERSION_MINOR}" VERSION_LESS "4.4")
-    return() 
+    return()
 endif()
 
-#Get LVGL version
-idf_component_get_property(lvgl_ver lvgl__lvgl COMPONENT_VERSION)
-if(lvgl_ver EQUAL "")
-    idf_component_get_property(lvgl_ver lvgl COMPONENT_VERSION)
+# This component uses a CMake workaround, so we can compile esp_lvgl_port for both LVGL8.x and LVGL9.x
+# At the time of idf_component_register() we don't know which LVGL version is used, so we only register an INTERFACE component (with no sources)
```

Comment:
FWIW, you can add a dummy source file here, and then call `target_sources(${COMPONENT_LIB} PRIVATE ${PORT_PATH}/esp_lvgl_port.c` once you know LVGL version, later.

Reply:
This is a backup plan :D I like this solution better, although I'm not 100% sure about its future-proofness or maintainability...